### ISSUE TITLE
Library build with "ivy" does not render components with @Input variables that have a custom TS "type" as type

### ISSUE THREAD

**Issue by mrucelum** [angular/angular#37199]
# 🐞 bug report

### Affected Package
I guess the ivy compiler? 

### Is this a regression?
Well it worked before, but we didn't use the ivy compiler before. Not sure if this counts as regression or not. :)

### Description

I have an angular library which contains a component that renders a simple list of sub components. For each item in the list, there is should be a component shown that has an input as follows:

```
@Input() public viewType: ViewType;
```

while `ViewType` is a type defined as follows

```
export type ViewType = 'card' | 'list';
```

My library is compiled with ivy enabled (I am aware that it is not yet recommended to do that, but we are using the library only internally anyway and afaik that's only for compatibility reasons?). Unfortunately, with the library I am experiencing have the same issues described here https://github.com/angular/angular/issues/20931 so I cannot just disable ivy build. :/

If I use this library in my angular app it won't show any item (in case of `--prod` used).
It only starts showing the items if I either
- do not use `--prod` flag
- turn off the optimizer (in `angular.json`)
- do not build the lib with ivy

I do not get any error, it just does not show the items. So the list is completely empty.

The workaround I found is not using our defined type `ViewType` but instead copy and paste its value:
```
@Input() public viewType: 'card' | 'list';
```
This works fine but is not really what I would prefer. :/

## 🔬 Minimal Reproduction
https://github.com/mrucelum/type-for-input-app for the test app and https://github.com/mrucelum/type-for-input-lib for the example lib.

Clone https://github.com/mrucelum/type-for-input-app and then run `yarn install` and `yarn start:prod`.

When you open localhost:4200 you will see a blank page.

If you run `yarn start` instead you will see 
```
item1
item2
```
on the page.

## 🔥 Exception or Error
No exception thrown. 

## 🌍  Your Environment

**Angular Version:**
<pre><code>
Angular CLI: 9.1.6
Node: 12.16.0
OS: win32 x64

Angular: 9.1.7
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.901.6
@angular-devkit/build-angular     0.901.6
@angular-devkit/build-optimizer   0.901.6
@angular-devkit/build-webpack     0.901.6
@angular-devkit/core              9.1.6
@angular-devkit/schematics        9.1.6
@angular/cli                      9.1.6
@ngtools/webpack                  9.1.6
@schematics/angular               9.1.6
@schematics/update                0.901.6
rxjs                              6.5.5
typescript                        3.8.3
webpack                           4.42.0
</code></pre>

**Comment by mrucelum** [comment]
Hi, so sorry, completely missed that importing ViewType in the child component (which is defined in the parent) produces a recursion. Will try it again, but it's probably really the same as #35602.

**Comment by JoostK** [comment]
Ah, thanks for the confirmation. It's actually a valid point that a type-only import shouldn't necessarily contribute to a cycle being detected. How we deal with cycles is being reconsidered and type-only imports could definitely be taken into account. Perhaps only if the `import type` syntax is used, available since TS 3.8 I believe.

**Comment by mrucelum** [comment]
Thanks for the information - that sounds good! :)
